### PR TITLE
DPL Analysis: add Gandiva compiled expressions to filter tables

### PR DESCRIPTION
**Pull request by aalkin**
- [x] Include type information into expression tree representation
- [x] Test that type information is correctly set
- [x] Create a translator from operation specs to Gandiva expression tree
- [x] Test and benchmark table filtering with Gandiva

### CODE REVIEW

**Comment by aalkin** [comment]
@ktf Now the type information is correctly propagated (see test_Expressions.cxx)

**Comment by ktf** [comment]
Thanks. It looks like you are missing `<unordered_map>` header in Expressions.cxx (I would also use a std::vector<pair<BasicOp, std::string>> or even just a `std::string[256]`, but that's a different story). Also please fix the codacy issue.

File: Framework/Core/src/ExpressionHelpers.h [hunk 1]
```diff
@@ -10,18 +10,48 @@
 #ifndef O2_FRAMEWORK_EXPRESSIONS_HELPERS_H_
 #define O2_FRAMEWORK_EXPRESSIONS_HELPERS_H_
 #include "Framework/Expressions.h"
+#include "gandiva/filter.h"
+
 #include <vector>
 #include <iosfwd>
 
+using namespace arrow;
```
Comment (ktf) [inline Framework/Core/src/ExpressionHelpers.h]: In principle we should avoid `using namespace` in headers...

Comment (aalkin) [reply Framework/Core/src/ExpressionHelpers.h]: It is an internal header, so I just saved myself some typing. I will remove the `using` statements when this is no longer WIP.

**Pull request merged**